code/week2/operations.cpp

```diff
@@ -0,0 +1,25 @@
+int add(int a, int b) {
+    return a + b;
+}
+
+int subtract(int a, int b) {
+    return add(a, -b);
+}
+
+int multiply(int a, int b) {
+    int accum = 0;
+    for(int i = 0; i < a; i++) {
```

Comment:
While 'we shouldn't be optimizing' is normally the case for example slides, this is a bit extreme. This 5 line cpp function can be replaced by a _single_ line of x86. The easiest assembly implementation of this function is simpler and easier to understand than this cpp function.

This sets a dangerous precedent because:

1) this implies cpp does not have native multiplication
2) It encourages roundabout, convoluted solutions when simple, obvious solutions are present
3) (if someone already knows cpp): it makes RJ look like amateurs 
4) It's a waste of student time to parse and understand this function (ie: what's going on here, is this just multiply? Am I missing something? There has to be a reason it's implemented with adds... What am I missing?).

I would suggest actually making a function that has a purpose, rather than making a  purely trivial example. Maybe implement a exponent function instead, using a wrapper around multiply? While `std::pow` does exist, at least this can be excused as an 'alternate implementation' of `std::pow`, rather than a re-implementation of a cpp primitive. Even better, maybe implement a looping factorial (which, afaik, isn't in `std`).